### PR TITLE
Rails 6.1 upgrade

### PR DESCRIPTION
**Pull request by LindseySaari**
## Description of change
This PR upgrades to the latest Rails version 6.1. There were various deprecation and breaking changes introduced between the 6.0 and 6.1 upgrade. Listed below are the changes that were addressed in the PR:

* ActiveModel Error Objects
    * New in 6.1, ActiveModel Error is now an object. ex: my_object.errors is now an array of objects vs a hash
    * Various sytax updates for custom error handling across the entire repository. 
    * Example errors[:some_attribute] << vs errors.add(:some_attribute]
* Fixture Paths
    * Removed the prepended files/ directory from file_fixture_uploads calls in the code. If fixtures are placed within the /files directory, the prepended /files directory is auto configured. 
    * app/controllers/concerns/exception_handling.rb has to use the explicit to_h method instead of to_hash due to object.errors now being an array

* Active Storage Blob migrations
    * These are new for Rails 6.1 and will need to go out with the gem upgrade (currently failing danger)
    * Some of the specs were failing with `NoMethodError (undefined method 'service_name' for #<ActiveStorage::Blob:0x0000558b5edfbeb8>)`. db migrations were needed for the ActiveStorage blobs (specifically to add the service_name column) Used the solution suggested [https://github.com/rails/rails/issues/40895#issuecomment-748811171](https://)

* Private headers method rename
    * Issue with private method `'headers'` in various controllers. Renamed the methods to `request_headers`
    * Believe that the private headers method was overlapping/overwriting the controller built in ‘headers’ method ex: headers['X-Session-Expiration'] 

* Environment config deprecation change
    * config/environments/test.rb 
    * DEPRECATION WARNING:` action_view.raise_on_missing_translations` is deprecated and will be removed in Rails 6.2. Set `i18n.raise_on_missing_translations` instead.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#18862
department-of-veterans-affairs/va.gov-team#22271

### CODE REVIEW

File: modules/appeals_api/spec/controllers/v1/contestable_issues_controller_spec.rb
Comment (LindseySaari) [inline modules/appeals_api/spec/controllers/v1/contestable_issues_controller_spec.rb:12]: This was split up to avoid rubocop line length

File: app/controllers/concerns/exception_handling.rb
Comment (LindseySaari) [inline app/controllers/concerns/exception_handling.rb:83]: errors are no longer a hash (now an object) so the explicit method call

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb
Comment (LindseySaari) [inline modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb:96]: Need to call explicit to_h method here too

File: modules/vaos/spec/models/cancel_form_spec.rb
Comment (LindseySaari) [inline modules/vaos/spec/models/cancel_form_spec.rb:13]: keys is deprecated, instead use attribute_names

**Comment by LindseySaari** [comment]
@cbtkc This is ready for review, it's just red because there's /db changes in the same PR

**Review by thilton-oddball — COMMENTED** [review]
This all looks good to me. The only two comments I have are that it looks like there are 59 rubocop violations that are all auto-correctable and then I wasn't sure if we want to split out the migrations into a separate PR or if they both have to be in this same PR.

**Review by kreek — APPROVED** [review]
Looks good for the mobile api change

**Comment by LindseySaari** [comment]
> This all looks good to me. The only two comments I have are that it looks like there are 59 rubocop violations that are all auto-correctable and then I wasn't sure if we want to split out the migrations into a separate PR or if they both have to be in this same PR.

Yes, I think that the migrations/schema changes will have to go out in this same PR because they're new in 6.1 and there was a failing spec related to the migration changes too. I will take a look at the rubocop violations.

**Review by cbtkc — APPROVED** [review]
Looks good!

File: Gemfile
Comment (ericboehs) [inline Gemfile:40]:
```suggestion
gem 'rails', '~> 6.1.3'
```

I'd recommend locking just to the patch.

File: app/models/concerns/redis_form.rb
Comment (ericboehs) [inline app/models/concerns/redis_form.rb:56]: Before the method would return `errors[:form]`. Now it's returning the enumerable. I don't know if this will affect anything but here's how you should be able to do what you're asking without doing the gross `v = foo; f.each {}; v` dance.

```suggestion
      JSON::Validator.fully_validate(VetsJsonSchema::SCHEMAS[self.class::FORM_ID], parsed_form).tap do |v|
        errors.add(:form, v.to_s)
      end
```

File: app/models/saved_claim.rb
Comment (ericboehs) [inline app/models/saved_claim.rb:89]: `tap`?

File: app/models/concerns/temp_form_validation.rb
Comment (ericboehs) [inline app/models/concerns/temp_form_validation.rb:22]: `tap`?

File: app/models/saved_claim/education_benefits/va_1990s.rb
Comment (ericboehs) [inline app/models/saved_claim/education_benefits/va_1990s.rb:11]: `tap`?

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb
Comment (ericboehs) [inline modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb:41]: I think `.slice` would work here instead of this craziness.

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb
Comment (ericboehs) [reply modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb:97]: Does this still need to happen?

**Review by ericboehs — COMMENTED** [review]
It would be good to go through [RailsDiff](https://railsdiff.org/6.0.0/6.1.3.2) and make sure we have all the changes add.

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb
Comment (LindseySaari) [reply modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb:97]: We're now iterating over an array of error objects

Comment (LindseySaari) [reply modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements_controller.rb:97]: See line 97

File: app/models/concerns/redis_form.rb
Comment (ericboehs) [inline app/models/concerns/redis_form.rb:53]: Each still returns an iterator instead of the original object.

```suggestion
      JSON::Validator.fully_validate(VetsJsonSchema::SCHEMAS[self.class::FORM_ID], parsed_form).tap do |v|
```

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb
Comment (ericboehs) [inline modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb:39]: Still feel like this should work:
```suggestion
    request.headers.slice EXPECTED_HEADERS
```

**Comment by LindseySaari** [comment]
@ericboehs I had to revert the .tap changes, which caused test failures

File: modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb
Comment (LindseySaari) [reply modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/base_contestable_issues_controller.rb:39]: This change caused some test failures @ericboehs, I reverted back to the original

**Comment by ericboehs** [comment]
> @ericboehs I had to revert the .tap changes, which caused test failures

Dang. :(

**Comment by LindseySaari** [comment]
@cbtkc @thilton-oddball @rileyanderson Can we get a re-review on this please? Also, still waiting on a re-review from @department-of-veterans-affairs/vfs-vaos

**Review by rileyanderson — APPROVED** [review]
LGTM 🚀

**Review by cbtkc — APPROVED** [review]
This looks good to me. Great work

**Comment by LindseySaari** [comment]
@department-of-veterans-affairs/vfs-vaos can we get a review on this again please! Thanks! @kreek

**Pull request merged**
